codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -113,6 +115,17 @@ public Http2ConnectionEncoder encoder() {
     }
 
     /**
+     * Internal testing use only!
+     */
+    ByteToMessageDecoder byteDecoder() {
+        return byteDecoder;
+    }
+
+    private boolean prefaceSent() {
+        return byteDecoder != null && byteDecoder.prefaceSent();
```

Comment:
In the current structure, yes. `byteDecoder` lifetime follows the handler's lifetime (created in `handlerAdded` and dereferenced in `handlerRemoved`).  However we could change the lifetime of this object to be the lifetime of `Http2ConnectionHandler`, but I didn't do this because needed to check this is only relevant for a brief time at the beginning of the connection (and then it is no longer checked).